scheduler/src/cook/rest/api.clj

```diff
@@ -2134,26 +2207,35 @@
                                                                           (into [])))}]
                          :else
-                         (let [groups (mapv #(validate-and-munge-group (db conn) %) groups)
-                               effective-pool-name (or (:pool/name pool) (config/default-pool))
-                               jobs (mapv #(validate-and-munge-job
-                                             (db conn)
-                                             effective-pool-name
-                                             user
-                                             task-constraints
-                                             gpu-enabled?
-                                             (set (map :uuid groups))
-                                             %
-                                             :override-group-immutability?
-                                             override-group-immutability?) jobs)
-                               {:keys [status message]} (submission-plugin/plugin-jobs-submission
-                                                          jobs
-                                                          effective-pool-name)]
+                         (let [groups (mapv #(validate-and-munge-group db %) groups)
+                               job-pool-name-maps
+                               (mapv
+                                 (fn [job]
+                                   (let [effective-pool-name
+                                         (pool-name->effective-pool-name pool-name job)
+                                         validated-and-munged-job
+                                         (validate-and-munge-job
+                                           db
+                                           effective-pool-name
+                                           user
+                                           task-constraints
+                                           gpu-enabled?
+                                           (set (map :uuid groups))
+                                           job
+                                           :override-group-immutability?
+                                           override-group-immutability?)]
+                                     {:job validated-and-munged-job
+                                      :pool-name effective-pool-name
+                                      :pool-name-from-submission pool-name}))
+                                 jobs)
+                               jobs (map :job job-pool-name-maps)
+                               {:keys [status message]}
+                               (submission-plugin/plugin-jobs-submission job-pool-name-maps)]
                            ; Does the plugin accept the submission?
                            (if (= :accepted status)
                              [false {::groups groups
                                      ::jobs jobs
```

Comment:
Is this ::jobs variable now dead?

Reply:
There was one place still referencing `::jobs`, which I've now changed to `::job-pool-name-maps`, and I removed this `::jobs`.